### PR TITLE
Use persistent MAC address instead clone / don't touch default

### PR DESCRIPTION
**Pull request by pvizeli**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/host/network.py [hunk 1]
```diff
@@ -279,15 +284,15 @@ def from_dbus_interface(inet: NetworkInterface) -> Interface:
                 inet.connection.ipv4.nameservers,
             )
             if inet.connection and inet.connection.ipv4
-            else None,
```
Comment (agners) [inline supervisor/host/network.py:282]: It feels to me nicer to return None, but if its a hassle for frontend maybe returning an empty is easier? @ludeeus

File: supervisor/dbus/payloads/interface_update.tmpl [hunk 2]
```diff
@@ -61,7 +61,7 @@
     ,
     '802-3-ethernet':
         {
-            'assigned-mac-address': <'stable'>
+            'assigned-mac-address': <'preserve'>
```
Comment (agners) [inline supervisor/dbus/payloads/interface_update.tmpl:64]: Is there a reason why we set this in first place? Can we not set it at all and leave it up to NM?

Comment (pvizeli) [reply supervisor/dbus/payloads/interface_update.tmpl:64]: Different NM working differently based on his version. The platform should work everywhere exactly in the same way and eliminate customizing. Also if we update NM on OS, it should not change his behavior after the update.

File: supervisor/host/network.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/host/network.py:282]: Not a hassle, but this is better IMO.

**Review by ludeeus — APPROVED** [review]

File: supervisor/host/network.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/host/network.py:282]: I agree, as we not represent NM on our abstracted interface, it's better to show what it currently is

**Pull request merged**